components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/service/impl/HTTPEventPublisherImpl.java

```diff
@@ -86,63 +82,101 @@ public boolean canHandleEvent(EventContext eventContext) throws EventPublisherEx
         }
     }
 
-    private void makeAsyncAPICall(SecurityEventTokenPayload eventPayload, EventContext eventContext)
-            throws HTTPAdapterException {
-
-        ClientManager clientManager = HTTPAdapterDataHolder.getInstance().getClientManager();
+    private void makeAsyncAPICall(SecurityEventTokenPayload eventPayload, EventContext eventContext) {
 
         for (Webhook webhook : activeWebhooks) {
             String url = webhook.getEndpoint();
             String secret = webhook.getSecret();
+            sendWithRetries(eventPayload, eventContext, url, secret,
+                    HTTPAdapterDataHolder.getInstance().getClientManager().getMaxRetries());
+        }
+    }
 
-            HttpPost request = clientManager.createHttpPost(url, eventPayload, secret);
-
-            logPublishingEvent(url, eventContext, eventContext.getEventUri());
-
-            final long requestStartTime = System.currentTimeMillis();
+    private void sendWithRetries(SecurityEventTokenPayload eventPayload, EventContext eventContext,
+                                 String url, String secret, int retriesLeft) {
 
-            CompletableFuture<HttpResponse> future = clientManager.executeAsync(request);
+        ClientManager clientManager = HTTPAdapterDataHolder.getInstance().getClientManager();
+        final HttpPost request;
+        try {
+            request = clientManager.createHttpPost(url, eventPayload, secret);
+        } catch (HTTPAdapterException e) {
+            printPublisherDiagnosticLog(eventContext, eventPayload, url,
+                    HTTPAdapterConstants.LogConstants.ActionIDs.PUBLISH_EVENT, DiagnosticLog.ResultStatus.FAILED,
+                    "Failed to construct HTTP request for HTTP adapter publish.");
+            log.debug("Error constructing HTTP request for HTTP adapter publish. No retries will be attempted.", e);
+            return;
+        }
 
-            future.thenAccept(response -> handleAsyncResponse(
-                            response, request, requestStartTime, eventContext, url, eventContext.getEventUri()))
-                    .exceptionally(ex -> {
+        printPublisherDiagnosticLog(eventContext, eventPayload, url,
+                HTTPAdapterConstants.LogConstants.ActionIDs.PUBLISH_EVENT, DiagnosticLog.ResultStatus.SUCCESS,
+                "Publishing event data to endpoint.");
+
+        final long requestStartTime = System.currentTimeMillis();
+        final String correlationId =
+                request.getFirstHeader(HTTPAdapterConstants.Http.CORRELATION_ID_REQUEST_HEADER).getValue();
```

Comment:
Lets make null safe for 
`request.getFirstHeader(HTTPAdapterConstants.Http.CORRELATION_ID_REQUEST_HEADER)`